### PR TITLE
fix in-place geometric pmf

### PR DESCRIPTION
**Pull request by min-jean-cho**
See https://github.com/pytorch/pytorch/pull/37984#discussion_r1059548320

cc @svekars @carljparker

### CODE REVIEW

**Review by lezcano — COMMENTED** [review]
Sure, but what does the implementation in core do? Does it implement the standardSure, but what does the implementation in core do? Does it implement the standard geometric distribution or the one described in the docs?

**Comment by min-jean-cho** [comment]
> Sure, but what does the implementation in core do? Does it implement the standard geometric distribution or the one described in the docs?

It implements the standard geometric distribution.
`P{X=i} = p * q^(i-1), i >= 1, where q = 1 -p`

This derivation here, https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/TransformationHelper.h#L149-L153, is derived from the above PMF. Otherwise, the implementation here would be wrong.

**Review by lezcano — APPROVED** [review]
Makes sense! Thank you for the catch!

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -2094,7 +2094,7 @@ def add_docstr_all(method, docstr):
 
 .. math::
 
-    f(X=k) = p^{k - 1} (1 - p)
+    f(X=k) = (1 - p)^{k - 1} (p)
```
Comment (lezcano) [inline torch/_tensor_docs.py:2097]: actually
```suggestion
    f(X=k) = (1 - p)^{k - 1} p
```

**Pull request closed without merging**